File: Languages/Javascript/rand_avg.js
// Hello, World!
console.log("Hello, World!");

const fs = require('fs');
const path = require('path');

function generateRandomNumbers() {
    const count = 1000;
    const outputDir = "../rand_avg output";
    const outputFile = path.join(outputDir, "random_numbers.txt");
    
    // Create output directory
    if (!fs.existsSync(outputDir)) {
        fs.mkdirSync(outputDir, { recursive: true });
    }
    
    // Generate 1000 random numbers
    const randomNumbers = [];
    let sum = 0;
    
    for (let i = 0; i < count; i++) {
        const num = Math.floor(Math.random() * 1000);
        randomNumbers.push(num);
        sum += num;
    }
    
    // Calculate mean
    const mean = sum / count;
    
    // Save to file
    try {
        const content = randomNumbers.join('\n');
        fs.writeFileSync(outputFile, content);
        
        console.log("Generated 1000 random numbers");
        console.log(`Mean: ${mean.toFixed(2)}`);
        console.log(`Saved to: ${outputFile}`);
    } catch (error) {
        console.log("Could not create output file");
    }
}

generateRandomNumbers();

File: Languages/Python/rand_avg.py
print("Hello, World!")

import random
import os

def generate_random_numbers():
    # Generate 1000 random numbers from 0 to 999
    random_numbers = [random.randint(0, 999) for _ in range(1000)]
    
    # Create output directory if it doesn't exist
    output_dir = "../rand_avg output"
    os.makedirs(output_dir, exist_ok=True)
    
    # Save to file
    output_file = os.path.join(output_dir, "random_numbers.txt")
    with open(output_file, 'w') as f:
        for num in random_numbers:
            f.write(f"{num}\n")
    
    # Calculate mean
    mean = sum(random_numbers) / len(random_numbers)
    
    print(f"Generated 1000 random numbers")
    print(f"Mean: {mean:.2f}")
    print(f"Saved to: {output_file}")

if __name__ == "__main__":
    generate_random_numbers()
